test/es7/asyncawait-syntax.js

```diff
@@ -128,6 +128,16 @@ var tests = [
             var d = async (a, b) => { };
         }
     },
+    {
+        name: "[no LineTerminator here] after `async` in grammar",
+        body: function () {
+            assert.throws(function () { eval("async\nfunction af() { }"); }, ReferenceError, "AsyncFunctionDeclaration", "'async' is undefined");
```

Comment:
Should have better descriptive errors than these ones?
eg. 'throw' has nice error if it has new lines between the expression.